libtock/ble.c

```diff
@@ -26,16 +26,15 @@ int ble_start_passive_scan(uint8_t *data, uint8_t max_len,
   if (data == NULL || callback == NULL) {
     return TOCK_FAIL;
   } else {
-    int err;
 
-    err = subscribe(BLE_DRIVER_NUMBER, BLE_SCAN_SUB, callback, NULL);
-    if (err < TOCK_SUCCESS)
-      return err;
+    int sub_err = subscribe(BLE_DRIVER_NUMBER, BLE_SCAN_SUB, callback, NULL);
+    if (sub_err < TOCK_SUCCESS)
+      return sub_err;
 
-    err =
-      allow(BLE_DRIVER_NUMBER, BLE_CFG_SCAN_BUF_ALLOW, (void *)data, max_len);
-    if (err < TOCK_SUCCESS)
-      return err;
+    allow_rw_return_t allow_err =
+      allow_readwrite(BLE_DRIVER_NUMBER, BLE_CFG_SCAN_BUF_ALLOWRW, (void *)data, max_len);
+    if (!allow_err.success)
+      return allow_err.error;
 
     return command(BLE_DRIVER_NUMBER, BLE_SCAN_CMD, 1, 0);
```

Comment:
```suggestion
    syscall_return_t res = command2(BLE_DRIVER_NUMBER, BLE_SCAN_CMD, 1, 0);
      if (res.type == TOCK_SYSCALL_SUCCESS) {
        return TOCK_SUCCESS;
      } else {
        return tock_error_to_rcode(res.data[0]);
      }
```

basically this should be done for all calls to command (though those that return values are of course a little different, see ADC for an example)

Reply:
Done